features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin_new/source/src/app/components/BotDetection/EmailConfig/AddEmail.jsx

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React, { useReducer } from 'react';
+import API from 'AppData/api';
+import PropTypes from 'prop-types';
+import Joi from '@hapi/joi';
+import TextField from '@material-ui/core/TextField';
+import DialogContentText from '@material-ui/core/DialogContentText';
+import { FormattedMessage } from 'react-intl';
+import { makeStyles } from '@material-ui/core/styles';
+import FormDialogBase from 'AppComponents/AdminPages/Addons/FormDialogBase';
+import Alert from 'AppComponents/Shared/Alert';
+
+const useStyles = makeStyles((theme) => ({
+    error: {
+        color: theme.palette.error.dark,
+    },
+}));
+
+/**
+ * Reducer
+ * @param {JSON} state State
+ * @returns {Promise}.
+ */
+function reducer(state, { field, value }) {
+    return {
+        ...state,
+        [field]: value,
+    };
+}
+
+/**
+ * Render a pop-up dialog to add an email address
+ * @param {JSON} props .
+ * @returns {JSX}.
+ */
+function AddEmail(props) {
+    const {
+        updateList, icon, triggerButtonText, title, emailList,
+    } = props;
+    const classes = useStyles();
+    const initialState = {
+        email: '',
+
+    };
+
+    const [state, dispatch] = useReducer(reducer, initialState);
```

Comment:
I think it's not worth having a reducer, just to keep a single state variable (`name='email'`), Shall we use `useState` instead? It will make the code cleaner WDYT ? 😃

Reply:
no dispatchers, not reducers simple 

```
const [email, setEmail] = useState();
```
and use the `setEmail` function in `onChange` handler

Reply:
Fixed with commit: https://github.com/wso2/carbon-apimgt/pull/8591/commits/814f59930b3d3e7e8507a02109f852c3906e686a